### ISSUE TITLE
Get apikey function

### ISSUE THREAD

**Issue by Lop010** [GaspardCulis/elevenlabs-unleashed#15]
Hi, im trying to an existing script that just uses the normal elevenlabs api with requests. I'd like to continue to use my existing function without changing it too much. Would it be possible to only get a apikey from this module?

**Comment by Lop010** [comment]
tried using this `name, password, api_key = create_account()` but i got this error `Exception: Captcha not checked in time`

**Comment by GaspardCulis** [comment]
Hi, please refer to [this comment](https://github.com/GaspardCulis/elevenlabs-unleashed/issues/12#issuecomment-1870913795) explaining the current state of the project

**Comment by Lop010** [comment]
hey i found this project maybe hope its helpful for you
https://github.com/NopeCHALLC/nopecha-extension

**Comment by Lop010** [comment]
wouldn't be the worst solution imo, would be better than nothing

**Comment by GaspardCulis** [comment]
> You could just load the extension using selenium https://stackoverflow.com/a/52420595

That's already what's being done, but the [hektCaptcha extension](https://github.com/Wikidepia/hektCaptcha-extension) is no longuer maintained, that's the issue.

<br>

> this project already use Selenium, one possible way is to temporarly pop-up the Chrome window and ask the user to solve the captcha manually for the session.

Not ideal, we might as well just create the accounts manually.

<br>

> hey i found this project maybe hope its helpful for you
> https://github.com/NopeCHALLC/nopecha-extension
> (only 100 solves per day, better than nothing tho)

The 100 solves a day put me off, but it actually works very well, I might use it works very well, but they do not give the .crx in their releases, has to be built manually or scraped from the webstore.

**Comment by XenocodeRCE** [comment]
> > You could just load the extension using selenium https://stackoverflow.com/a/52420595
> 
> That's already what's being done [here](https://github.com/GaspardCulis/elevenlabs-unleashed/blob/906792cbc0007e5734c70d8d69f0287438d38106/src/elevenlabs_unleashed/account.py#L149), but the [hektCaptcha extension](https://github.com/Wikidepia/hektCaptcha-extension) is no longuer maintained, that's the issue.
> 
> 
> > this project already use Selenium, one possible way is to temporarly pop-up the Chrome window and ask the user to solve the captcha manually for the session.
> 
> Not ideal, we might as well just create the accounts manually.
> 
> 
> > hey i found this project maybe hope its helpful for you
> > https://github.com/NopeCHALLC/nopecha-extension
> > (only 100 solves per day, better than nothing tho)
> 
> The 100 solves a day put me off, but it actually works very well, I might use it, but they do not give the .crx in their releases, has to be built manually or scraped from the webstore.

Is the registration captcha systematic ? Can't we be white-listed for a certain amount of time ?

**Comment by GaspardCulis** [comment]
> Is the registration captcha systematic ? Can't we be white-listed for a certain amount of time ? 

Yes it is, I don't think the 11Labs team went into the trouble of creating a whitelist

**Comment by expeditofreitas** [comment]
switch to undetected_chromedriver, it will ignore captcha

**Comment by GaspardCulis** [comment]
> switch to undetected_chromedriver, it will ignore captcha

Won't help, 11Labs asks for a Captcha rergardless of the browser you are using.

**Comment by expeditofreitas** [comment]
> > switch to undetected_chromedriver, it will ignore captcha
> 
> Won't help, 11Labs asks for a Captcha rergardless of the browser you are using.
> 
> EDIT: Tried it to be sure, not helping...

work
https://youtu.be/jeDE5uirBlM

**Comment by GaspardCulis** [comment]
Okay MB I'm dumb I wasn't actually using the test version with undetected_chromedriver on my testing. Actually amazing works like a charm, you seem to have code working with their new website, do you want to submit a [PR](https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github) for that ?

**Comment by expeditofreitas** [comment]
> Okay MB I'm dumb I wasn't actually using the test version with undetected_chromedriver on my testing. Actually amazing works like a charm, you seem to have code working with their new website, do you want to submit a [PR](https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github) for that ?

you can also simply use an existing crhomer session
options.add_argument('user-data-dir=C:\\Users\\Exped\\AppData\\Local\\Google\\Chrome\\User Data\\Profile 1') #you path
options.add_argument("--incognito")

**Comment by GaspardCulis** [comment]
I was talking about the code for submitting the 11Labs onboarding questions that you've figured out

**Comment by GaspardCulis** [comment]
Got everything working again in [afb0d97](https://github.com/GaspardCulis/elevenlabs-unleashed/commit/afb0d97f60421d1a5f2a4dec0d2a525736f6d1bd), but undetected_chromedriver fails to fill in the captcha 90% of the time for me

**Comment by AbhinavMishra32** [comment]
How can we enable filling captcha manually?

**Comment by GaspardCulis** [comment]
> How can we enable filling captcha manually?

Run the `create_account` function with `DEBUG` environment variable set to **true** to make the browser visible

**Comment by AbhinavMishra32** [comment]
> > How can we enable filling captcha manually?
> 
> Run the `create_account` function with `DEBUG` environment variable set to **true** to make the browser visible
> 
> 

I dont know how to do that😕 can you tell me how to set environment variable for a function? I had done `set DEBUG=1` before in terminal to see the browser window but that didnt work...

**Comment by GaspardCulis** [comment]
Saw you fork my repo just delete [that line](https://github.com/AbhinavMishra32/elevenlabs-unleashed/blob/f11835b176a7fcb4bcda6494f338cac736259261/src/elevenlabs_unleashed/account.py#L103)

**Comment by self-muson** [comment]
hello im new to programming but know something. 
the problem im stack now its: 
1. "Exception: Captcha not checked in time"
2. i dont understand how is chromedriver work, i have broser and chromedriver installed co C:/ and added to PATH but browser doesnt open to solve captcha

**Comment by GaspardCulis** [comment]
Hi, currently, the captcha solving works rarely. I still haven't found a reliable up-to-date captcha solver

**Comment by self-muson** [comment]
> Hi, currently, the captcha solving works rarely. I still haven't found a reliable up-to-date captcha solver

hello sir. so as i assume, the project is not working well right now - bcoz of captcha sloving?

**Comment by Sholabumi** [comment]
Kinda new to this, so if possible can someone write a descriptive message on how I am suppose to run and use the code? (.P.S. read description and full page of what was posted on how to di it. Didn't help me much)
